lightning_pose/utils/predictions.py

```diff
@@ -354,12 +360,31 @@ def predict_single_video(
     # get loader
     predict_loader = vid_pred_class()
 
-    preds = trainer.predict(
-        model=model,
-        ckpt_path=ckpt_file,
-        dataloaders=predict_loader,
-        return_predictions=True,
-    )
+    # initialize prediction handler class
+    pred_handler = PredictionHandler(cfg=cfg, data_module=data_module, video_file=video_file)
+
+    # use a different function for now to return heatmaps
+    if save_heatmaps:
+        if predict_loader.do_context:
+            batch_size = cfg.dali.context.predict.batch_size
+        else:
+            batch_size = cfg.dali.base.predict.sequence_length
+        keypoints, confidences, heatmaps = _predict_frames(
+            cfg=cfg, model=model, dataloader=predict_loader, n_frames_=pred_handler.frame_count,
+            batch_size=batch_size, return_heatmaps=True)
+        preds = [(torch.tensor(keypoints), torch.tensor(confidences))]
+        if heatmaps is not None:
+            heatmaps_file = preds_file.replace(".csv", "_heatmaps.npy")
+            os.makedirs(os.path.dirname(heatmaps_file), exist_ok=True)
+            np.save(heatmaps_file, heatmaps)
+
+    else:
```

Comment:
please add a comment -- `# vanilla prediction, returns keypoints and their associated confidences.`